typing/includemod.ml

```diff
@@ -134,21 +134,94 @@ module Error = struct
 
 end
 
-type mark =
+module Directionality = struct
+
+
+  type mark =
   | Mark_both
   | Mark_positive
-  | Mark_negative
   | Mark_neither
 
-let negate_mark = function
-  | Mark_both -> Mark_both
-  | Mark_positive -> Mark_negative
-  | Mark_negative -> Mark_positive
-  | Mark_neither -> Mark_neither
+  type pos =
+    | Strictly_positive
+      (** Strictly positive positions are notable for tools since they are the
+          the case where we match a implementation definition with an interface
+          declaration. Oherwise in the positive case we are matching
+          declatations inside functor arguments at even level of nesting.*)
+    | Positive
+    | Negative
+
+
+(**
+   When checking inclusion, the [Directionality.t] type tracks the
+   subtyping direction at the syntactic level.
 
-let mark_positive = function
-  | Mark_both | Mark_positive -> true
-  | Mark_negative | Mark_neither -> false
+   The [posivity] field is used in the [cmt_declaration_dependencies] to
+   distinguish between directed and undirected edges, and to avoid recording
+   matched declarations twice.
```

Comment:
I think you mean [pos] rather than [positivity]. It is unclear to me how the [pos] type "distinguishes between directed and undirected edges". (Or what this means in the first place. I suppose the "edges" are the dependencies, but what does it mean for a dependency to be undirected? Maybe this is better explained somewhere else, but then you could refer to that place.)

Edit: in fact maybe you don't need to document cmt_declaration_dependencies here anymore: now that the `pos` information is necessarily to correctly "mark" the submodules, it is used for strictly more than that.